bcda/api.go

```diff
@@ -275,6 +276,13 @@ func jobStatus(w http.ResponseWriter, r *http.Request) {
 	case "Failed":
 		responseutils.WriteError(&fhirmodels.OperationOutcome{}, w, http.StatusInternalServerError)
 	case "Completed":
+		// If the job should be expired, but the cleanup job hasn't run for some reason, still respond with 410
```

Comment:
okay this makes sense but if the cleanup hasn't run then we have bigger issues. So im thinking, "is it correct logic to put this here?" But from our users perspective they don't and shouldn't care about our internal jobs so this makes sense in these cases. 1) the job is completed but we're notifying them that it's going to be expiring. 2) The cleanup job didn't run for some reason but they're still correctly notified that it's expired with the 410 status. Plus I think the plan is to have some kind of cloud watch monitoring to make sure those files were successfully archived so hopefully we won't run into that latter scenario :)

Reply:
I hear you and I agree, but by the same token, this code can only help us conform the software to our expectations of how it works so I think it should stay.  It is like the elevator with multiple levels of fail safes.